### DISCUSSION TITLE
Subscribe to events (listener pattern) in .jar

### DISCUSSION THREAD

**Discussion by mrwhy-orig** [ikvmnet/ikvm#454] [Q&A]
Hi, 
is there an example how to subscribe to events in the java library? Would it be straight forward? Just implementing the listener pattern in the .Net application? Or is there something I have to take care about? 
Is there anywhere an up to date documentation or should I stick to the sourceforge copied one?

Background:
I have an existing Java application and I have to reuse it in C#. I have to use a certain communication library only available in C# and put it on top of the old Java application. But I would need to be able to be informed on certain events from the java application.

Thank you!

**Comment by wasabii** [comment]
"the Java lbrary?"

Depends on the actual Java library. All IKVM does is let you run and access the classes in the Java from .NET. Anything you'd want to actually do with that library is going to be up to the library.

**Reply by mrwhy-orig** [reply]
Well not really the java libraray but an example how to use the listener or an event pattern from java to c#. 
But from your answer I assume i can use the existing java application, extend it, subscribe to and get notified in c#.  Can I use Java interfaces in C# with IKVM ?

**Reply by wasabii** [reply]
Same way you would in Java. I have no idea what specifically you're asking about. Java classes turn into .NET classes.
